### ISSUE TITLE
Pipeline scheduling should never block on resource checks

### ISSUE THREAD

**Issue by vito** [concourse/concourse#2438]
# Feature Request

## What challenge are you facing?

The root cause of "my pipeline stopped scheduling" is pretty often to do with resource checks that the scheduler performs for pending builds that don't have pinned inputs.

## A Modest Proposal

We should somehow decouple pipeline scheduling from resource checking. Pipeline scheduling is supposed to be cheap and able to happen multiple times a minute. Doing checks in the middle increases risk of hanging and just generally slows things down.

**Comment by pivotal-jwinters** [comment]
related to #1954

**Comment by clarafu** [comment]
After some discussion, we will remove the force resource checks from the manually triggered builds and instead separate the responsibilities of the manually trigger button on the job into two separate buttons:
- a button that will force a check for all your resources that are inputs to your job
- another button that will manually trigger a new build with the current versions of each resource

This issue will be blocked on #3403.

**Comment by cirocosta** [comment]
Hey @clarafu , do we also plan to have a separate flag for `check-resource`?

**Comment by vito** [comment]
Hmm I'd like to think about this a bit more. Having a button that triggers without checking feels a little wishy washy - I could see it being misinterpreted as either 'trigger with newest available versions' (today's behavior) or 'retrigger THIS BUILD with same versions' (#413), but in reality it does neither.

What if we had the check button short-circuit the scheduling loop, so that if there *are* new versions, a new build will be queued up promptly, and then just went ahead and finally did #413 for the second button?

**Comment by vito** [comment]
@cirocosta Also yeah, we'd need to think about that command - maybe even split it up so there's no chance for ambiguity. Something like `fly retrigger-build` and `fly check-and-schedule-job` (terrible name - I feel like we need a verb here for the 'check and schedule soonish' action).

**Comment by vito** [comment]
Ok, after talking about this in IPM here's the idea:

* Introduce a 'sync job' button (to be named) that does the following:
    * Run a `check` for all resources that go into the job
    * After the `check`s complete, hits an API endpoint to "kick" scheduling for the job.
      * We'll need to think about this in the API, I think.
    * ...Somehow waits for a new build, which may or may not happen if the scheduler determines there's no new build to trigger.
      * We'll need to think about this in the API, I think.

* Do #413 

...we're running low on time so I'll fill in more details later.

**Comment by vito** [comment]
Ok. So. That all led to a whole build page redesign.

Here's a short(er)-term plan for getting checking removed from the scheduler:

1. For manually triggered builds, instead of doing a check in-place, just wait for all the input resources to have a `last_checked` that is after the build's creation time. If they're not, just leave the build pending (similar to how we handle `serial: true`).
1. That's it! Technically all user needs are met here; the build won't run until it has the latest versions.
1. That would mean the user has to wait for up to a minute (or their `check_every`) before the build runs. So, we could just use `LISTEN`/`NOTIFY` to have the scheduler notify the resource scanner and short-circuit its scheduling loop.

We talked a bit about adding a scan queue, which would be a bit more accurate rather than relying on timestamps, but we figure that can be added later when we want to clean up the check API endpoints, which currently call the resource scanner synchronously.

**Comment by xtremerui** [comment]
> 3. That would mean the user has to wait for up to a minute (or their `check_every`) before the build runs. So, we could just use `LISTEN`/`NOTIFY` to have the scheduler notify the resource scanner and short-circuit its scheduling loop.
 
Since there is https://github.com/concourse/concourse/issues/3508 should we move the `LISTEN/NOTIFY` part to there? @vito
